### PR TITLE
Add GitLab CI trigger in GitHub checks 

### PR DESCRIPTION
**Pull request by nvda-mesharma**
Mirror repo on internal Gitlab repo
Trigger CI 
Report status

### CODE REVIEW

File: .github/workflows/trigger_ci.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ jobs:
     needs: mirror_repo
     runs-on: self-hosted
     steps:
-      - name: step2
+      - name: step 2
```
Comment (rmccorm4) [inline .github/workflows/trigger_ci.yml:17]:
```suggestion
      - name: Trigger Pipeline
```

File: .github/workflows/trigger_ci.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+on:
+  pull_request:
+jobs:
+  mirror_repo:
+    environment: GITLAB
+    runs-on: self-hosted
+    steps:
+      - name: Mirror Repository
```
Comment (rmccorm4) [inline .github/workflows/trigger_ci.yml:8]: The repository is already mirrored, right? Is this "updating" the mirror or forcing a sync?

Comment (nvda-mesharma) [reply .github/workflows/trigger_ci.yml:8]: yes, this enforces the mirroring to be done immediately

Comment (rmccorm4) [reply .github/workflows/trigger_ci.yml:8]: Can we call it "Sync Mirror Repository" then? Sorry for the nits, just want it to be crystal clear.

Comment (mc-nv) [reply .github/workflows/trigger_ci.yml:8]: I'm trying to understand the intention. 
CI can be triggered without this change, isn't?

Comment (rmccorm4) [reply .github/workflows/trigger_ci.yml:8]: Not automatically from github side. Nor would it report back a pipeline on the github PR status checks without this change

Comment (nvda-mesharma) [reply .github/workflows/trigger_ci.yml:8]: > Not automatically from github side. Nor would it report back a pipeline on the github PR status checks without the corresponding gitlab PR's changes
That's right.

Comment (rmccorm4) [inline .github/workflows/trigger_ci.yml:8]:
```suggestion
      - name: Sync Mirror Repository
```

**Review by rmccorm4 — APPROVED** [review]

**Pull request merged**